### PR TITLE
Adding Adherent list in Municipal Chief space

### PR DESCRIPTION
**Pull request by ktherage**

### CODE REVIEW

File: src/Repository/AdherentRepository.php [hunk 1]
```diff
@@ -770,4 +770,20 @@ public function refresh(Adherent $adherent): void
     {
         $this->getEntityManager()->refresh($adherent);
     }
+
+    public function findForMunicipalChief(Adherent $municipalChief): iterable
```
Comment (ottaviano) [inline src/Repository/AdherentRepository.php]: `: array` ?

File: templates/municipal_chief/adherent/list.html.twig [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                </thead>
+                <tbody>
+                {% for user in results %}
+                    <tr class="referent__item">
+                        <td class="adherent-identity">
+                            <div class="adherent-name">
+                                {{ user.lastName }} {{ user.firstName }}
+                            </div>
+                            <div class="adherent-caracs">
+                                {% if user.gender %}
```
Comment (ottaviano) [inline templates/municipal_chief/adherent/list.html.twig]: you could use 
```twig
{{ user.gender ? 'common.gender.'~ user.gender|trans : '~' }}
```

File: templates/municipal_chief/adherent/list.html.twig [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+{% block stylesheets %}
+    {{ parent() }}
+    <link href="{{ asset('select2/select2.min.css') }}" rel="stylesheet">
+{% endblock %}
+
+{% block municipal_chief_content %}
+    <div class="datagrid__pre-table b__nudge--bottom-larger">
+        <h3>Adhérents</h3>
+        <span class="datagrid__count">
+            <span>{{ results.count |number_format(0, ',', ' ') }}</span> sur {{ results.totalItems |number_format(0, ',', ' ') }} adhérent{{ total_count > 1 ? 's' : '' }}
```
Comment (thomasbisignani) [inline templates/municipal_chief/adherent/list.html.twig]: Can you glue the filters twig please ? 
`{{ results.count |number_format ..}}` => `{{ results.count|number_format ... }}`

File: src/Repository/AdherentRepository.php [hunk 4]
```diff
@@ -770,4 +771,25 @@ public function refresh(Adherent $adherent): void
...
         $this->getEntityManager()->refresh($adherent);
     }
+
+    public function findPaginatedForMunicipalChief(
+        Adherent $municipalChief,
+        int $page = 1,
+        int $maxItemPerPage = 10
+    ): ApiPaginator {
+        if (!$municipalChief->isMunicipalChief()) {
+            return [];
```
Comment (ottaviano) [inline src/Repository/AdherentRepository.php]: will not work with `: ApiPaginator` typehint.
what about renaming the method in:
`findPaginatedForInseeCodes(['insee1', 'insee2'], ...)` 
and remove this check:
`if (!$municipalChief->isMunicipalChief()) {`?

File: templates/municipal_chief/adherent/list.html.twig [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+{% block stylesheets %}
+    {{ parent() }}
+    <link href="{{ asset('select2/select2.min.css') }}" rel="stylesheet">
+{% endblock %}
+
+{% block municipal_chief_content %}
+    <div class="datagrid__pre-table b__nudge--bottom-larger">
+        <h3>Adhérents</h3>
+        <span class="datagrid__count">
+            <span>{{ results.count|number_format(0, ',', ' ') }}</span> sur {{ results.totalItems|number_format(0, ',', ' ') }} adhérent{{ total_count > 1 ? 's' : '' }}
```
Comment (ottaviano) [inline templates/municipal_chief/adherent/list.html.twig]: `total_count` => `results.totalItems`

File: src/Controller/EnMarche/MunicipalChiefController.php [hunk 6]
```diff
@@ -291,4 +292,21 @@ private function handleApplicationRequestTagsRequest(
...
+        Request $request,
+        UserInterface $municipalChief,
+        AdherentRepository $adherentRepository
+    ): Response {
+        $this->disableInProduction();
+        $paginator = $adherentRepository->findPaginatedForMunicipalChief($municipalChief, $request->query->getInt('page'));
+
+        return $this->render('municipal_chief/adherent/list.html.twig', [
+            'results' => $paginator,
+            'total_count' => $paginator->count(),
```
[the remark this answers is not in the document]
Comment (ktherage) [reply src/Controller/EnMarche/MunicipalChiefController.php]: i just forgot to remove it. What a PR sniper you are @ottaviano :wink:.

File: src/Controller/EnMarche/MunicipalChiefController.php [hunk 7]
```diff
@@ -291,4 +292,23 @@ private function handleApplicationRequestTagsRequest(
...
+    public function adherentsListAction(
+        Request $request,
+        UserInterface $municipalChief,
+        AdherentRepository $adherentRepository
+    ): Response {
+        $this->disableInProduction();
+        $inseeCodes = $municipalChief->getMunicipalChiefManagedArea()->getCodesAsString();
+        $paginator = $adherentRepository->findPaginatedForInseeCodes($inseeCodes, $request->query->getInt('page'));
+
+        dump($paginator);
```
Comment (ottaviano) [inline src/Controller/EnMarche/MunicipalChiefController.php]: 🙆‍♂

File: src/Repository/AdherentRepository.php [hunk 8]
```diff
@@ -770,4 +771,22 @@ public function refresh(Adherent $adherent): void
     {
         $this->getEntityManager()->refresh($adherent);
     }
+
+    public function findPaginatedForInseeCodes(
+        string $inseeCodes,
```
Comment (ottaviano) [inline src/Repository/AdherentRepository.php]: it should be an array here, with the string you cannot be sure that it follows a good format for `FIND_IN_SET` function (`val1,val2,valN`).

**Pull request merged**